data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/ElasticsearchAccessor.java

```diff
@@ -149,7 +153,8 @@ public CreateScrollResponse createScroll(final CreateScrollRequest createScrollR
         SearchResponse<ObjectNode> searchResponse;
 
         try {
-            searchResponse = elasticsearchClient.search(SearchRequest.of(request -> request
+            searchResponse = elasticsearchClientRefresher.get()
```

Comment:
Since this change is somewhat intrusive to the code, and we don't currently have integration tests for this source, just curious what testing you did to verify no regression?

Reply:
Yes. I run a smoke test with pipeline:

```
pipeline_configurations:
  aws:
    secrets:
      secret1:
        secret_id: my-es-secret
        region: us-east-1
        sts_role_arn: "arn:aws:iam::xxxx:role/ecs-test-role"
      secret2:
        secret_id: my-self-managed-os
        region: us-east-1
        sts_role_arn: "arn:aws:iam::xxxx:role/ecs-test-role"
opensearch-migration-pipeline:
  source:
    opensearch:
      hosts: [ "https://localhost:9200" ]
      username: "${{aws_secrets:secret2:username}}"
      password: "${{aws_secrets:secret2:password}}"
  processor:
    - date:
        from_time_received: true
  sink:
    - opensearch:
        hosts: [ "https://search-test-domain-r3dk3xkywa3vqqdzdqrddq3kyq.us-east-1.es.amazonaws.com" ]
        username: "${{aws_secrets:secret1:username}}"
        password: "${{aws_secrets:secret1:password}}"
        index: "test-migration"
```